### PR TITLE
[v2] various fixes

### PR DESCRIPTION
**Pull request by waynz0r**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| API breaks?     | no
| Deprecations?   | no
| Related tickets | 
| License         | Apache 2.0

### CODE REVIEW

**Review by stoader — APPROVED** [review]

File: api/v1alpha1/common.proto [hunk 1]
```diff
@@ -538,16 +538,20 @@ message Replicas {
     // +kubebuilder:validation:Minimum=0
     google.protobuf.Int32Value count = 1 [(gogoproto.wktpointer) = true];
 
-    // Standard Kubernetes minimum replicas configuration
+    // min is the lower limit for the number of replicas to which the autoscaler
+    // can scale down.
+    // min and max both needs to be set the turn on autoscaling.
```
Comment (Laci21) [inline api/v1alpha1/common.proto]:
```suggestion
    // min and max both need to be set the turn on autoscaling.
```

File: api/v1alpha1/common.proto [hunk 2]
```diff
@@ -538,16 +538,20 @@ message Replicas {
...
     // +kubebuilder:validation:Minimum=0
     google.protobuf.Int32Value min = 2 [(gogoproto.wktpointer) = true];
 
-    // Standard Kubernetes maximum replicas configuration
-    // +kubebuilder:validation:Minimum=0
+    // max is the upper limit for the number of replicas to which the autoscaler can scale up.
+    // min and max both needs to be set the turn on autoscaling.
```
Comment (Laci21) [inline api/v1alpha1/common.proto]:
```suggestion
    // min and max both need to be set the turn on autoscaling.
```

File: api/v1alpha1/common.proto [hunk 3]
```diff
@@ -538,16 +538,20 @@ message Replicas {
...
+    // It cannot be less that min.
```
Comment (Laci21) [inline api/v1alpha1/common.proto]:
```suggestion
    // It cannot be less than min.
```

File: controllers/istiocontrolplane_controller.go [hunk 4]
```diff
@@ -189,7 +189,7 @@ func (r *IstioControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	}
 
 	updateErr := components.UpdateStatus(ctx, r.Client, icp, components.ConvertConfigStateToReconcileStatus(servicemeshv1alpha1.ConfigState_Available), "")
-	if updateErr != nil {
+	if updateErr != nil && !k8serrors.IsNotFound(updateErr) {
```
Comment (Laci21) [inline controllers/istiocontrolplane_controller.go:192]: thx

File: internal/assets/manifests/istio-discovery/templates/autoscale.yaml [hunk 5]
```diff
@@ -19,6 +19,6 @@ spec:
   - type: Resource
     resource:
       name: cpu
-      targetAverageUtilization: {{ .Values.pilot.cpu.targetAverageUtilization }}
+      targetAverageUtilization: {{ default 80 .Values.pilot.cpu.targetAverageUtilization }}
```
Comment (Laci21) [inline internal/assets/manifests/istio-discovery/templates/autoscale.yaml]: shouldn't we add the default to `values.yaml`?

File: internal/assets/manifests/istio-meshgateway/templates/autoscale.yaml [hunk 6]
```diff
@@ -18,6 +18,6 @@ spec:
     - type: Resource
       resource:
         name: cpu
-        targetAverageUtilization: {{ $gateway.replicas.targetCPUUtilizationPercentage }}
+        targetAverageUtilization: {{ default 80 $gateway.replicas.targetCPUUtilizationPercentage }}
```
Comment (Laci21) [inline internal/assets/manifests/istio-meshgateway/templates/autoscale.yaml]: same

**Review by Laci21 — APPROVED** [review]
just a few nits

**Review by Laci21 — APPROVED** [review]

**Pull request merged**